core/src/main/scala/kafka/api/FetchResponse.scala

```diff
@@ -48,6 +49,36 @@ object FetchResponsePartitionData {
 
 case class FetchResponsePartitionData(error: Short = Errors.NONE.code, hw: Long = -1L, messages: MessageSet) {
   val sizeInBytes = FetchResponsePartitionData.headerSize + messages.sizeInBytes
+
+  def toMessageFormat(toMagicValue: Byte): FetchResponsePartitionData = {
```

Comment:
Wouldn't it be better to put this conversion in KafkaApis or directly in MessageSet? I would prefer to keep FetchResponse and FetchResponsePartitionData as simple as possible and focused on the serialization.